sds/sds.go

```diff
@@ -0,0 +1,559 @@
+		return nil, err
+	}
+
+	Debug("Wrapping outgoing message %v", messageId)
+
+	wg := sync.WaitGroup{}
+	var resp = C.allocResp(unsafe.Pointer(&wg))
+	defer C.freeResp(resp)
+
+	cMessageId := C.CString(string(messageId))
+	defer C.free(unsafe.Pointer(cMessageId))
+
+	var cMessagePtr unsafe.Pointer
+	if len(message) > 0 {
+		cMessagePtr = C.CBytes(message) // C.CBytes allocates memory that needs to be freed
+		defer C.free(cMessagePtr)
+	} else {
+		cMessagePtr = nil
+	}
+	cMessageLen := C.size_t(len(message))
+
+	wg.Add(1)
+	C.cGoWrapOutgoingMessage(rm.rmCtx, cMessagePtr, cMessageLen, cMessageId, resp)
+	wg.Wait()
+
+	if C.getRet(resp) == C.RET_OK {
+		resStr := C.GoStringN(C.getMyCharPtr(resp), C.int(C.getMyCharLen(resp)))
+		if resStr == "" {
+			Debug("Received empty res string for messageId: %v", messageId)
+			return nil, nil
+		}
+		Debug("Successfully wrapped message %s", messageId)
+
+		parts := strings.Split(resStr, ",")
+		bytes := make([]byte, len(parts))
+
+		for i, part := range parts {
+			n, err := strconv.Atoi(strings.TrimSpace(part))
+			if err != nil {
+				panic(err)
+			}
+			bytes[i] = byte(n)
+		}
+
+		return bytes, nil
+	}
+
+	errMsg := "error WrapOutgoingMessage: " + C.GoStringN(C.getMyCharPtr(resp), C.int(C.getMyCharLen(resp)))
+	Error("Failed to wrap message %v: %v", messageId, errMsg)
+
+	return nil, errors.New(errMsg)
+}
+
+func (rm *ReliabilityManager) UnwrapReceivedMessage(message []byte) (*UnwrappedMessage, error) {
+	if rm == nil {
+		err := errors.New("reliability manager is nil")
+		Error("Failed to unwrap received message %v", err)
+		return nil, err
+	}
+
+	wg := sync.WaitGroup{}
+	var resp = C.allocResp(unsafe.Pointer(&wg))
+	defer C.freeResp(resp)
+
+	var cMessagePtr unsafe.Pointer
+	if len(message) > 0 {
+		cMessagePtr = C.CBytes(message) // C.CBytes allocates memory that needs to be freed
+		defer C.free(cMessagePtr)
+	} else {
+		cMessagePtr = nil
+	}
+	cMessageLen := C.size_t(len(message))
+
+	wg.Add(1)
+	C.cGoUnwrapReceivedMessage(rm.rmCtx, cMessagePtr, cMessageLen, resp)
+	wg.Wait()
+
+	if C.getRet(resp) == C.RET_OK {
+		resStr := C.GoStringN(C.getMyCharPtr(resp), C.int(C.getMyCharLen(resp)))
+		if resStr == "" {
+			Debug("Received empty res string")
+			return nil, nil
+		}
+		Debug("Successfully unwrapped message")
+
+		unwrappedMessage := UnwrappedMessage{}
+		err := json.Unmarshal([]byte(resStr), &unwrappedMessage)
+		if err != nil {
+			Error("Failed to unmarshal unwrapped message")
+			return nil, err
+		}
+
+		return &unwrappedMessage, nil
+	}
+
+	errMsg := "error UnwrapReceivedMessage: " + C.GoStringN(C.getMyCharPtr(resp), C.int(C.getMyCharLen(resp)))
+	Error("Failed to unwrap message: %v", errMsg)
+
+	return nil, errors.New(errMsg)
+}
+
+// MarkDependenciesMet informs the library that dependencies are met
+func (rm *ReliabilityManager) MarkDependenciesMet(messageIDs []MessageID) error {
+	if rm == nil {
+		err := errors.New("reliability manager is nil")
```

Comment:
In errors I think is interesting give as much context as possible :)
```suggestion
		err := errors.New("reliability manager is nil in MarkDependenciesMet")
```